module/data/chat-message/request-message-data.mjs

```diff
@@ -87,27 +118,49 @@ export default class RequestMessageData extends ChatMessageDataModel {
    */
   static async #handleRequest(event, target) {
     const actor = fromUuidSync(target.closest("[data-uuid]").dataset.uuid);
-    const result = await CONFIG.DND5E.requests[this.handler](actor, this.parent, this.data);
-    if ( result instanceof ChatMessage ) {
+    const result = await CONFIG.DND5E.requests[this.handler](actor, this.parent, this.data, { event });
+    if ( (result instanceof ChatMessage) && !result.getFlag("dnd5e", "requestResult") ) {
```

Comment:
Workaround for https://github.com/foundryvtt/foundryvtt/issues/13067. We try to add the flags during message creation now, and I've added an associated `createChatMessage` hook.